utils.js

```diff
@@ -0,0 +1,22 @@
+// class ProductArray {
```

Comment:
this file is looking good. This could be where you define your class and then you wouldn't need to export your functions, they wold just be methods on your class.